modules/wiki/wiki.py

```
@@ -260,13 +260,12 @@ async def _callback(msg: Bot.MessageSession):
                                 i_msg_lst.append(Plain(session.locale.t('wiki.message.invalid_section.select')))
                                 i_msg_lst.append(Plain(session.locale.t('message.reply.prompt')))
 
-                                async def _callback(msg: Bot.MessageSession):
-                                    display = msg.as_display(text_only=True)
-                                    if isint(display):
-                                        if int(display) <= len(forum_data) - 1:
-                                            await query_pages(session, title=forum_data[display]['text'], start_wiki_api=r.info.api)
+                            async def _callback(msg: Bot.MessageSession):
+                                display = msg.as_display(text_only=True)
+                                if isint(display) and int(display) <= len(forum_data) - 1:
```

Ruff (0.8.0)

Function definition does not bind loop variable `forum_data`

(B023)